### ISSUE TITLE
Add a material property to chose texture orientation relative to the camera viewpoint

### ISSUE THREAD

**Issue by swiss-knight** [CesiumGS/cesium#9185]
Hello,
I want to be able to decide the orientation of a texture image applied to an [entity.polygon](https://cesium.com/docs/cesiumjs-ref-doc/Entity.html#polygon) relative to the camera viewpoint.

Now it seems totally random (or I didn't get how the texture orientation is chosen); sometime the text on an image is readable, and sometime it's flipped left<->right.

I'm currently applying the texture as follow:

```
entity.polygon.material = new Cesium.ImageMaterialProperty({
    image: './image.jpg',
    alpha: 0.5
});
```

In addition, the alpha value doesn't apply. It would be great to be able to add transparency to the texture.
Thanks.

**Comment by OmarShehata** [comment]
Note that there is no `alpha` property. See: https://cesium.com/docs/cesiumjs-ref-doc/ImageMaterialProperty.html?classFilter=ImageMaterialProperty. Have you tried setting `transparent: true` and setting a color with an alpha?

**Comment by dzungpng** [comment]
Closing this issue as we have not heard from the original poster in a while. Please feel free to open it if you want to continue the discussion.

**Comment by chaijinsong** [comment]
the same problem, i use
```js
target.material = new Cesium.ImageMaterialProperty({
   image: url
})
```
but the img is not right rotation
![image](https://user-images.githubusercontent.com/25416773/100454500-fb884800-30f7-11eb-8fed-351eedffca09.png)
my source picture is
![image](https://user-images.githubusercontent.com/25416773/100454601-31c5c780-30f8-11eb-9743-4fae82596636.png)

the picture is rotated, it's wrong!
first i think the problem is about of picture width, so i test this situation. the result is not about width
dose anyone have this problem ？ please give me some advise！

**Comment by chaijinsong** [comment]
@dzungpng

**Comment by dzungpng** [comment]
I tried directly apply the material to the polygon on creation and the orientation seems fine:
[Sandcastle](https://sandcastle.cesium.com/#c=rVPRbpswFP2VK55IRE1gygul0aTsJVKnTVm1J0uVA7fMm7Ej29AkVf59BkJCsmhtpT7he+859rnniJppqDk+o4Y7kPgMczS8KsnPtudTL2vruZKWcYmaeqNbKqmsHU9jZpksBDpqdwdBabnlaAjLc/+FSgCeJ0C9JeYneAC7hcxxAxH1ggZznCTQcgAypXTOJbNokl7SskeRJ63KL1hoRON3eICbKJqQSdCX8bC4iSYTMj2Wn9ysO48OvU5PAtGhLt3DmjORDC1ZlKzAr4fJd63WqO3Wf+lvBd7Mm2UJCduzCTvm470q1ONcCaXJ73VBvY6yHzXfvXty33lKZTim8mgG6VWcJ/M/Ge+RcHj0leAecGMrfTW9+GPTi4fpTc/Ti4fpTS/TO6X1hsUvEo+DsxDG4cCTnVLlg/IvHGpgXuClxm4FznpVn3m5VtpCpYXvNFgs16LZP1xV2R+0JDMtsYGm4ZCa5rx2Xt9d+dMgE8wYN3mqhPjBd0i9WRo6/D9UoZjzu/hWoxZs28B+RbP7rkkISUNXXmdapcSK6Yub/wI)

![image](https://user-images.githubusercontent.com/38567338/100797037-82d30580-33ef-11eb-9e2c-1737adbaf308.png)


But if we apply the material after we created the polygon, I get a flipped/blank texture (similar to what @chaijinsong is experiencing):
[Sandcastle](https://sandcastle.cesium.com/#c=rVNda9swFP0rFz85xZVjj7y4bhhkL4WOja7sSVBU+9bTJktBkt0mJf99khWnThb2AXuyru45V+eeg3umoef4jBquQeIzrNDwriVfh7uYRtVQr5S0jEvUNJpdUUll73gaK8tkI9BRwwyC0nLL0RBW1/ErlQC8LoBGd1i/wRPY3sgaXyCjUeIxh04BAwegUkrXXDKLphgl3Y0o8qRV+wEbjWjigAe4zLI5mSdjmU+Ly2w+J4tD+c71wnm2vwt6Csj2dXoRvq0ToDkTxdSam5Y1+HHf+azVGrXdxK/jdOC+75cmJB3OJg3Mh1vVqIeVEkqT7+uGRoGym12k/rRzj++Cu1QeLCGjhuN8fifiXwQMD/p1/xDgPb7YTp9NMf+/KebTFBfHKebTFBenKb6l9RernySfJ0cR+EQOnmyVau9VfOKQh0VJVBq7EbgcVb3n7VppC50WsdNgsV0Lv3/62FU/0JLKDEQPLdMptax577y+PvPHQSWYMa7z1AnxhW+RRssydfhfqEIx53fzqUct2MbDvmXL23BJCClTV55nWqXEI9Mnk38C)

![image](https://user-images.githubusercontent.com/38567338/100796993-751d8000-33ef-11eb-81d6-ff3eda1ac7b7.png)

**Comment by OmarShehata** [comment]
@dzungpng you have a bug in your code. The following line is incorrect:

```
rectangle.material = new Cesium.ImageMaterialProperty({
   image: "../images/Cesium_Logo_Color.jpg"
});
```

It should instead be `rectangle.rectangle.material = `, because the first "rectangle" is the entity itself, not the rectangle. 

If the goal is to rotate the material applied on the rectangle this can be done with the `stRotation`, see: https://cesium.com/docs/cesiumjs-ref-doc/RectangleGraphics.html?classFilter=RectangleGr#stRotation

I think what the original user is asking for is a way to decide which way the image is facing, which I don't think there exists a way to control/it might be intentionally set up to be facing depending on whether the vertices are clockwise/anti-clockwise.

**Comment by dzungpng** [comment]
@OmarShehata  I see. Thanks for the clarification!

**Comment by swiss-knight** [comment]
> @dzungpng you have a bug in your code. The following line is incorrect:
> 
> ```
> rectangle.material = new Cesium.ImageMaterialProperty({
>    image: "../images/Cesium_Logo_Color.jpg"
> });
> ```
> 
> It should instead be `rectangle.rectangle.material = `, because the first "rectangle" is the entity itself, not the rectangle.
> 
> If the goal is to rotate the material applied on the rectangle this can be done with the `stRotation`, see: https://cesium.com/docs/cesiumjs-ref-doc/RectangleGraphics.html?classFilter=RectangleGr#stRotation
> 
> I think what the original user is asking for is a way to decide which way the image is facing, which I don't think there exists a way to control/it might be intentionally set up to be facing depending on whether the vertices are clockwise/anti-clockwise.

Yes, vertices order was the first way I explored, but I didn't succeeded (see sandcastle hereunder), both way (clockwise and anti-clockwise were giving the same results).

But as it has to be relative to the camera viewpoint, the problem should be stated an other way.
So I would personally rephrase this to the fact that it is basically a choice on the normal direction relative to the camera position ; e.g. the normal of to the rectangle is positive when it's going toward the half-space where the camera position actually is, or positive the other way.

This use case would be nice to have for static features. But for moving features, it will obviously switch the image texture at the exact moment the camera is entering the other half-space defined by the polygon plane.

Here's an example:

```
var viewer = new Cesium.Viewer("cesiumContainer");

/* // Camera 1 is in one half-space defined by the polygon plane
viewer.camera.setView({
    destination: Cesium.Cartesian3.fromDegrees(10.75031095, 48.3140656, 575),
    orientation: {
        heading : Cesium.Math.toRadians(1*300),   // heading
        pitch   : Cesium.Math.toRadians(1*-16),  // pitch
        roll    : Cesium.Math.toRadians(0*0)    // roll
    }
});
/**/

 // Camera 2 is in other half-space defined by the polygon plane
viewer.camera.setView({
    destination: Cesium.Cartesian3.fromDegrees(10.7420390, 48.317041, 575),
    orientation: {
        heading : Cesium.Math.toRadians(1*120),   // heading
        pitch   : Cesium.Math.toRadians(1*-16),  // pitch
        roll    : Cesium.Math.toRadians(0*0)    // roll
    }
});
/**/

const poly = {
    "type": "FeatureCollection",
    "name": "MyPolygon",
    "crs": {"type": "name",
            "properties": {
                "name": "urn:ogc:def:crs:OGC:1.3:CRS84"
    }},
    "features": [
        {"type": "Feature",
         "properties": {},
         "geometry": {
             "type": "Polygon",
             "coordinates": [
                /**/
                [ // clockwise-clockwise; negative rotation from the camera
                    [ 10.746038315853207, 48.314905418639555, 550 ], // Ul
                    [ 10.747038310965864, 48.315905422444722, 550 ], // Ur
                    [ 10.747500009924019, 48.315700009648104, 500 ], // Br
                    [ 10.746500009923748, 48.314700009648320, 500 ], // Bl
                    [ 10.746038315853207, 48.314905418639555, 550 ], // Ul
                ]
                /*
                [ // counter-clockwise; positive rotation from the camera
                    [ 10.746500009923748, 48.314700009648320, 500 ], // Bl
                    [ 10.747500009924019, 48.315700009648104, 500 ], // Br
                    [ 10.747038310965864, 48.315905422444722, 550 ], // Ur
                    [ 10.746038315853207, 48.314905418639555, 550 ], // Ul
                    [ 10.746500009923748, 48.314700009648320, 500 ], // Bl
                ]
                /**/
]}}]};


const displayfunc = async () => {
    try {
        const dataSource = await Cesium.GeoJsonDataSource.load(poly, {
            stroke: Cesium.Color.BLUE,
            strokeWidth: 3
        });
        
        const polygonalFrame = await viewer.dataSources.add(dataSource);
        const entities = polygonalFrame.entities.values;
        entities[0].polygon.material = new Cesium.ImageMaterialProperty({
            image: '../images/Cesium_Logo_Color.jpg',
            alpha: 0.5
        });
    }
    catch (e) {
        console.log("Error:", e);
    }
};

displayfunc();
```
=> https://sandcastle.cesium.com (v.1.77)

- First, try to switch from the two rotations senses by activating the other block of coordinates in the JSON object. it's not going to change anything
- Second, switch the camera. There we are... reversed logo.

I don't know how the normal is actually handled, I've the feeling that the texture is applied with a positive reading direction (if it contains text) when the rectangle is seen 'from above', because when the logo is reversed, if I only tilt the rectangle plane so that its upper face is facing the half-space containing the camera position, the texture is correct again.
